### PR TITLE
Feature/eu cert logic

### PR DESCRIPTION
**Pull request by ubamrein**
This Pull-Request marks the start of the transition to be fully compliant with "CertLogic" and the EU-Gateway format.

We introduce a function to downgrade the new format to the previously format, though, for simplicity, the file is generated once and statically stored.

For the v2 endpoint, the value-sets dynamically fetched from the gateway are injected. Further, we now allow the usage of CertLogic to evaluate to date-time expressions, to allow clients to present "validFrom" and "validTo" based on this logic.

### CODE REVIEW

File: ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        valueSetValues.add(fieldNameIterator.next());
+                    }
+                    valueSets.put(id, valueSetValues);
+                } catch (Exception ex) {
+                    continue;
+                }
+            }
+        }
+        var etag = EtagUtil.getSha1HashForStrings(strings.toArray(new String[0]));
+        if (request.checkNotModified(verificationRulesEtag + etag)) {
```
Comment (ubhaller) [inline ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2.java:89]: springboot not modified magic only allows gzip compression if weak etags are used. check if appending two weak etags is still considered a valid weak etag (`W/\"etag1\"W/\"etag2\"`)

File: ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        HashMap<String, ArrayList<String>> valueSets = new HashMap<>();
+        ArrayList<String> strings = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+        for (var id : allIds) {
+            var valueSet = valueSetDataService.findLatestValueSet(id);
+            if (valueSet != null) {
+                try {
+                    var entryJson = mapper.readTree(valueSet);
+                    strings.add(valueSet);
+                    var fieldNameIterator = entryJson.get("valueSetValues").fieldNames();
```
Comment (ubhaller) [inline ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2.java:77]: see other comment about empty iterator

File: ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                    var valueSetValues = new ArrayList<String>();
+                    while (fieldNameIterator.hasNext()) {
+                        valueSetValues.add(fieldNameIterator.next());
+                    }
+                    valueSets.put(id, valueSetValues);
+                } catch (Exception ex) {
+                    continue;
```
Comment (ubhaller) [inline ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2.java:84]: log exception

**Review by ubhaller — CHANGES_REQUESTED** [review]

File: ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesController.java [hunk 4]
```diff
@@ -41,9 +44,29 @@
...
+
+        InputStream verificationRulesFile =
+                new ClassPathResource("verificationRules.json").getInputStream();
         this.verificationRules = mapper.readValue(verificationRulesFile, Map.class);
-        this.verificationRulesEtag = EtagUtil.getSha1HashForFiles(verificationRulesFile.getPath());
+        this.verificationRulesEtag =
+                EtagUtil.getSha1HashForFiles("classpath:verificationRules.json");
+    }
+
+    private ArrayList<Map> mapV2RulesToV1(JsonNode template) throws IOException {
```
[the remark this answers is not in the document]
Comment (ubamrein) [reply ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesController.java:55]: yes, currently unused, but didn't want to delete the code I used to "transform" v2 (though it was manually)

**Review by ubhaller — APPROVED** [review]

**Pull request merged**